src/app/workflow/launch-third-party/launch-third-party.component.ts

```diff
@@ -321,4 +344,48 @@ export class LaunchThirdPartyComponent extends Base implements OnChanges, OnInit
     }
     return `Export this workflow version to ${platform}.`;
   }
+
+  launchToCodespace() {
+    const correctDevcontainerPath = this.computeCorrectDevcontainerPath(this.devcontainers);
+    if (correctDevcontainerPath) {
+      this.openNewCodespaceWindow(correctDevcontainerPath);
+    } else {
+      this.displayLaunchToCodespaceDialog(this.devcontainers.length > 0)
+        .afterClosed()
+        .subscribe((shouldLaunch: boolean) => {
+          if (shouldLaunch) {
+            this.openNewCodespaceWindow(undefined);
+          }
+        });
+    }
+  }
+
+  private displayLaunchToCodespaceDialog(hasDevcontainer: boolean) {
+    return this.dialog.open(LaunchToCodespaceDialogComponent, {
+      width: bootstrap4largeModalSize,
+      data: { entry: this.workflow, hasDevcontainer: hasDevcontainer },
+    });
+  }
+
+  private computeCorrectDevcontainerPath(devcontainers: SourceFile[]): string | undefined {
+    // Return the path, relative to root, of the devcontainer with a body that contains the absolute notebook path as a substring,
+    // or undefined if there is no such devcontainer
+    return devcontainers.find((file) => file.content.includes(this.selectedVersion.workflow_path))?.absolutePath?.substring(1);
+  }
+
+  private openNewCodespaceWindow(devcontainerPath: string | undefined) {
+    let url: string =
+      this.config.GITHUB_CODESPACES_IMPORT_URL +
+      '?hide_repo_select=true' +
+      '&ref=' +
+      this.selectedVersionNameAsQueryValue +
+      '&repo=' +
+      this.workflowOrganizationAsQueryValue +
+      '/' +
+      this.workflowRepositoryAsQueryValue;
+    if (devcontainerPath) {
+      url += '&devcontainer_path=' + this.encode(devcontainerPath);
+    }
+    window.open(url, '_blank');
```

Comment:
Our HTML links to external sites usually (should) have `noopener noreferrer`, should probably have here too.

Reply:
Done, https://stackoverflow.com/questions/40593632/use-window-open-but-block-use-of-window-opener